### ISSUE TITLE
Errors while installing latest version

### ISSUE THREAD

**Issue by orlagetty** [confluentinc/confluent-kafka-python#1784]
Description
===========
Errors while installing latest version (2.5.0). 
```
/mnt/tmp/pip-install-blxssofn/confluent-kafka/src/confluent_kafka/src/confluent_kafka.h:18:10: fatal error: Python.h: No such file or directory
     #include <Python.h>
```

confluent-kafka version: 2.5.0
OS: Amazon Linux 2023

How to reproduce
================

Executing command: `sudo python3 -m pip install confluent-kafka`

Error output
=========
```
Collecting confluent-kafka
  Using cached confluent-kafka-2.5.0.tar.gz (136 kB)
Using legacy 'setup.py install' for confluent-kafka, since package 'wheel' is not installed.
Installing collected packages: confluent-kafka
    Running setup.py install for confluent-kafka ... error
    ERROR: Command errored out with exit status 1:
     command: /bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/mnt/tmp/pip-install-blxssofn/confluent-kafka/setup.py'"'"'; __file__='"'"'/mnt/tmp/pip-install-blxssofn/confluent-kafka/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /mnt/tmp/pip-record-ntd03130/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.7m/confluent-kafka
         cwd: /mnt/tmp/pip-install-blxssofn/confluent-kafka/
    Complete output (67 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/confluent_kafka
    copying src/confluent_kafka/__init__.py -> build/lib.linux-x86_64-3.7/confluent_kafka
    copying src/confluent_kafka/deserializing_consumer.py -> build/lib.linux-x86_64-3.7/confluent_kafka
    copying src/confluent_kafka/error.py -> build/lib.linux-x86_64-3.7/confluent_kafka
    copying src/confluent_kafka/serializing_producer.py -> build/lib.linux-x86_64-3.7/confluent_kafka
    creating build/lib.linux-x86_64-3.7/confluent_kafka/_model
    copying src/confluent_kafka/_model/__init__.py -> build/lib.linux-x86_64-3.7/confluent_kafka/_model
    creating build/lib.linux-x86_64-3.7/confluent_kafka/_util
    copying src/confluent_kafka/_util/__init__.py -> build/lib.linux-x86_64-3.7/confluent_kafka/_util
    copying src/confluent_kafka/_util/conversion_util.py -> build/lib.linux-x86_64-3.7/confluent_kafka/_util
    copying src/confluent_kafka/_util/validation_util.py -> build/lib.linux-x86_64-3.7/confluent_kafka/_util
    creating build/lib.linux-x86_64-3.7/confluent_kafka/admin
    copying src/confluent_kafka/admin/__init__.py -> build/lib.linux-x86_64-3.7/confluent_kafka/admin
    copying src/confluent_kafka/admin/_acl.py -> build/lib.linux-x86_64-3.7/confluent_kafka/admin
    copying src/confluent_kafka/admin/_cluster.py -> build/lib.linux-x86_64-3.7/confluent_kafka/admin
    copying src/confluent_kafka/admin/_config.py -> build/lib.linux-x86_64-3.7/confluent_kafka/admin
    copying src/confluent_kafka/admin/_group.py -> build/lib.linux-x86_64-3.7/confluent_kafka/admin
    copying src/confluent_kafka/admin/_listoffsets.py -> build/lib.linux-x86_64-3.7/confluent_kafka/admin
    copying src/confluent_kafka/admin/_metadata.py -> build/lib.linux-x86_64-3.7/confluent_kafka/admin
    copying src/confluent_kafka/admin/_records.py -> build/lib.linux-x86_64-3.7/confluent_kafka/admin
    copying src/confluent_kafka/admin/_resource.py -> build/lib.linux-x86_64-3.7/confluent_kafka/admin
    copying src/confluent_kafka/admin/_scram.py -> build/lib.linux-x86_64-3.7/confluent_kafka/admin
    copying src/confluent_kafka/admin/_topic.py -> build/lib.linux-x86_64-3.7/confluent_kafka/admin
    creating build/lib.linux-x86_64-3.7/confluent_kafka/avro
    copying src/confluent_kafka/avro/__init__.py -> build/lib.linux-x86_64-3.7/confluent_kafka/avro
    copying src/confluent_kafka/avro/cached_schema_registry_client.py -> build/lib.linux-x86_64-3.7/confluent_kafka/avro
    copying src/confluent_kafka/avro/error.py -> build/lib.linux-x86_64-3.7/confluent_kafka/avro
    copying src/confluent_kafka/avro/load.py -> build/lib.linux-x86_64-3.7/confluent_kafka/avro
    creating build/lib.linux-x86_64-3.7/confluent_kafka/kafkatest
    copying src/confluent_kafka/kafkatest/__init__.py -> build/lib.linux-x86_64-3.7/confluent_kafka/kafkatest
    copying src/confluent_kafka/kafkatest/verifiable_client.py -> build/lib.linux-x86_64-3.7/confluent_kafka/kafkatest
    copying src/confluent_kafka/kafkatest/verifiable_consumer.py -> build/lib.linux-x86_64-3.7/confluent_kafka/kafkatest
    copying src/confluent_kafka/kafkatest/verifiable_producer.py -> build/lib.linux-x86_64-3.7/confluent_kafka/kafkatest
    creating build/lib.linux-x86_64-3.7/confluent_kafka/schema_registry
    copying src/confluent_kafka/schema_registry/__init__.py -> build/lib.linux-x86_64-3.7/confluent_kafka/schema_registry
    copying src/confluent_kafka/schema_registry/avro.py -> build/lib.linux-x86_64-3.7/confluent_kafka/schema_registry
    copying src/confluent_kafka/schema_registry/error.py -> build/lib.linux-x86_64-3.7/confluent_kafka/schema_registry
    copying src/confluent_kafka/schema_registry/json_schema.py -> build/lib.linux-x86_64-3.7/confluent_kafka/schema_registry
    copying src/confluent_kafka/schema_registry/protobuf.py -> build/lib.linux-x86_64-3.7/confluent_kafka/schema_registry
    copying src/confluent_kafka/schema_registry/schema_registry_client.py -> build/lib.linux-x86_64-3.7/confluent_kafka/schema_registry
    creating build/lib.linux-x86_64-3.7/confluent_kafka/serialization
    copying src/confluent_kafka/serialization/__init__.py -> build/lib.linux-x86_64-3.7/confluent_kafka/serialization
    creating build/lib.linux-x86_64-3.7/confluent_kafka/avro/serializer
    copying src/confluent_kafka/avro/serializer/__init__.py -> build/lib.linux-x86_64-3.7/confluent_kafka/avro/serializer
    copying src/confluent_kafka/avro/serializer/message_serializer.py -> build/lib.linux-x86_64-3.7/confluent_kafka/avro/serializer
    running build_ext
    building 'confluent_kafka.cimpl' extension
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/mnt
    creating build/temp.linux-x86_64-3.7/mnt/tmp
    creating build/temp.linux-x86_64-3.7/mnt/tmp/pip-install-blxssofn
    creating build/temp.linux-x86_64-3.7/mnt/tmp/pip-install-blxssofn/confluent-kafka
    creating build/temp.linux-x86_64-3.7/mnt/tmp/pip-install-blxssofn/confluent-kafka/src
    creating build/temp.linux-x86_64-3.7/mnt/tmp/pip-install-blxssofn/confluent-kafka/src/confluent_kafka
    creating build/temp.linux-x86_64-3.7/mnt/tmp/pip-install-blxssofn/confluent-kafka/src/confluent_kafka/src
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv
-fPIC -I/usr/include/python3.7m -c /mnt/tmp/pip-install-blxssofn/confluent-kafka/src/confluent_kafka/src/confluent_kafka.c -o build/temp.linux-x86_64-3.7/mnt/tmp/pip-install-blxssofn/confluent-kafka/src/confluent_kafka/src/confluent_kafka.o
    In file included from /mnt/tmp/pip-install-blxssofn/confluent-kafka/src/confluent_kafka/src/confluent_kafka.c:17:0:
    /mnt/tmp/pip-install-blxssofn/confluent-kafka/src/confluent_kafka/src/confluent_kafka.h:18:10: fatal error: Python.h: No such file or directory
     #include <Python.h>
              ^~~~~~~~~~
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/mnt/tmp/pip-install-blxssofn/confluent-kafka/setup.py'"'"'; __file__='"'"'/mnt/tmp/pip-install-blxssofn/confluent-kafka/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /mnt/tmp/pip-record-ntd03130/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.7m/confluent-kafka Check the logs for full command output.

```

Checklist
=========
Please provide the following information:

 - [ x] confluent-kafka-python and librdkafka version (`confluent_kafka.version()` and `confluent_kafka.libversion()`):
 - [ ] Apache Kafka broker version:
 - [ ] Client configuration: `{...}`
 - [ x] Operating system:
 - [x ] Provide client logs (with `'debug': '..'` as necessary)
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by TheVexy** [comment]
Installing the following new libs fixed it for me 
<img width="647" alt="image" src="https://github.com/user-attachments/assets/70e14753-43d7-4a74-91b6-24df6d215bde">

**Comment by orlagetty** [comment]
Python 3.7

**Comment by pranavrth** [comment]
Please upgrade pip first and try.

`python -m pip install --upgrade pip`

**Comment by orlagetty** [comment]
We have a bootstrap script that always runs when deploying the scripts and part of it is
`sudo python3 -m pip install --upgrade pip`

So I don't think it's that.

**Comment by pranavrth** [comment]
From logs I could see one more thing that `wheel` package is not installed. Please install `wheel` package first and then try to install `confluent-kafka`. Generally this is the problem with older pip versions. Just for understanding, please provide exact version of python and pip as well using `python3 -V` and `python3 -m pip -V` respectively.

```Using legacy 'setup.py install' for confluent-kafka, since package 'wheel' is not installed.```

**Comment by orlagetty** [comment]
![image](https://github.com/user-attachments/assets/22708fa7-35ef-4e4e-9e7e-a49ee90082ea)

**Comment by pranavrth** [comment]
Can you please install `wheel` first? I am not facing this issue with amazonlinux or 3.7.16 in my local.

`python3 -m pip install wheel`
`python3 -m pip install confluent-kafka`

If this doesn't work, please provide output of `python3 -m pip install -vvv confluent-kafka`.

**Comment by airballking** [comment]
I can somewhat confirm this bug. 

On an up-to-date Ubuntu 20.04 system with pip installed via apt (which leads to pip version 20.0.2) installing confluent-kafka fails with an error on wheels:

![grafik](https://github.com/user-attachments/assets/bf411569-03bc-4bd1-b981-ab64aed09992)

First Upgrading pip via itself, and then trying to install the newest version of confluent-kafka, resolves the issue:

![grafik](https://github.com/user-attachments/assets/c4d9ed2a-f97a-4d4d-a658-8c24e6140be0)

As another hint: Installing version 2.4.0 of confluent-kafka in the original environment with pip version 20.0.2 works fine:

![grafik](https://github.com/user-attachments/assets/b9d34385-ba64-4da2-94d7-3c71481537fe)

So, in conclusion some aspects of version 2.5.0 of confluent-kafka seem to break installation on older pip versions.

**Comment by pranavrth** [comment]
We have moved building manylinux wheels to manylinux_2_28 in this release. This might have caused something with old pip. It would be really helpful if you can provide verbose logs while installing using `python3 -m pip install -vvv confluent-kafka`. I am not able to replicate this issue in my local.

For the meantime, please fix the issue by upgrading the pip version to latest.

**Comment by wholesale-error** [comment]
I am also facing similar issue.
Recreating the issue:
`docker run -v ${pwd}:/var/task "amazon/aws-sam-cli-build-image-python3.9" /bin/sh -c "
    pip install --upgrade pip &&
    python -m pip install -vvv confluent-kafka; exit"`
    
Output of `python3 -m pip install -vvv confluent-kafka` as requested by @pranavrth 
`
Requirement already satisfied: pip in /var/lang/lib/python3.9/site-packages (23.0.1)
Collecting pip
  Downloading pip-24.2-py3-none-any.whl (1.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.8/1.8 MB 14.8 MB/s eta 0:00:00
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 23.0.1
    Uninstalling pip-23.0.1:
      Successfully uninstalled pip-23.0.1
Successfully installed pip-24.2
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
Using pip 24.2 from /var/lang/lib/python3.9/site-packages/pip (python 3.9)
Non-user install because site-packages writeable
Disabling truststore because Python version isn't 3.10+
Created temporary directory: /tmp/pip-build-tracker-vwcy5rg0
Initialized build tracking at /tmp/pip-build-tracker-vwcy5rg0
Created build tracker: /tmp/pip-build-tracker-vwcy5rg0
Entered build tracker: /tmp/pip-build-tracker-vwcy5rg0
Created temporary directory: /tmp/pip-install-7g7fjlcr
Created temporary directory: /tmp/pip-ephem-wheel-cache-f4j1e54f
1 location(s) to search for versions of confluent-kafka:
* https://pypi.org/simple/confluent-kafka/
Fetching project page and analyzing links: https://pypi.org/simple/confluent-kafka/
Getting page https://pypi.org/simple/confluent-kafka/
Found index url https://pypi.org/simple/
Looking up "https://pypi.org/simple/confluent-kafka/" in the cache
Request header has "max_age" as 0, cache bypassed
No cache entry available
Starting new HTTPS connection (1): pypi.org:443
https://pypi.org:443 "GET /simple/confluent-kafka/ HTTP/1.1" 200 77169
Updating cache with response from "https://pypi.org/simple/confluent-kafka/"
etag object cached for 1209600 seconds
Caching due to etag
Fetched page https://pypi.org/simple/confluent-kafka/ as application/vnd.pypi.simple.v1+json
  Found link https://files.pythonhosted.org/packages/b5/0a/2c4053a5ad38b71e74f606c7987513818f009e344cbc1cdadda723f0d251/confluent-kafka-0.9.1.1.tar.gz (from https://pypi.org/simple/confluent-kafka/), version: 0.9.1.1
  Found link https://files.pythonhosted.org/packages/2b/48/63278a8b857805f5459caa06f1bc881c5022e2c1c9e9e9751310d986b73e/confluent-kafka-0.9.1.2.tar.gz (from https://pypi.org/simple/confluent-kafka/), version: 0.9.1.2
  Found link https://files.pythonhosted.org/packages/92/23/78acb04c6660f32fbb8f789252e202898c0b3e0cea2ec4ac0913b7970188/confluent-kafka-0.9.2.tar.gz (from https://pypi.org/simple/confluent-kafka/), version: 0.9.2
  Found link https://files.pythonhosted.org/packages/bb/aa/4183a3ea74fe75a05a76a4ee97edd897f03a2bd2e25c76ff8c437717fb24/confluent-kafka-0.9.4.tar.gz (from https://pypi.org/simple/confluent-kafka/), version: 0.9.4
  Found link https://files.pythonhosted.org/packages/96/36/516a2b7f592376968296d10de10a20f1ce411e5ff24a86b1a23d5a5f4042/confluent-kafka-0.11.0.tar.gz (from https://pypi.org/simple/confluent-kafka/), version: 0.11.0
  Found link https://files.pythonhosted.org/packages/21/84/68b1dcae3d849d2ff07ea08ba1913c6699105ac64c01c50da69937111812/confluent-kafka-0.11.4.tar.gz (from https://pypi.org/simple/confluent-kafka/), version: 0.11.4
  ............
  ............
  ............
Skipping link: not a file: https://pypi.org/simple/confluent-kafka/
Given no hashes to check 42 links for project 'confluent-kafka': discarding no candidates
Collecting confluent-kafka
  Created temporary directory: /tmp/pip-unpack-qsm80orp
  Looking up "https://files.pythonhosted.org/packages/94/51/2e8f4048edbcde0346047dcd5228985ddcd0ef4d518f878321c07cc1b39c/confluent-kafka-2.5.0.tar.gz" in the cache
  No cache entry available
  No cache entry available
  Starting new HTTPS connection (1): files.pythonhosted.org:443
  https://files.pythonhosted.org:443 "GET /packages/94/51/2e8f4048edbcde0346047dcd5228985ddcd0ef4d518f878321c07cc1b39c/confluent-kafka-2.5.0.tar.gz HTTP/1.1" 200 136857
  Downloading confluent-kafka-2.5.0.tar.gz (136 kB)
  Ignoring unknown cache-control directive: immutable
  Updating cache with response from "https://files.pythonhosted.org/packages/94/51/2e8f4048edbcde0346047dcd5228985ddcd0ef4d518f878321c07cc1b39c/confluent-kafka-2.5.0.tar.gz"
  etag object cached for 1209600 seconds
  Caching due to etag
  Added confluent-kafka from https://files.pythonhosted.org/packages/94/51/2e8f4048edbcde0346047dcd5228985ddcd0ef4d518f878321c07cc1b39c/confluent-kafka-2.5.0.tar.gz to build tracker '/tmp/pip-build-tracker-vwcy5rg0'
  Running setup.py (path:/tmp/pip-install-7g7fjlcr/confluent-kafka_9a3c7c6a7dbb4764b3a4d26658ee059b/setup.py) egg_info for package confluent-kafka
  Created temporary directory: /tmp/pip-pip-egg-info-xd486t19
  Preparing metadata (setup.py): started
  Running command python setup.py egg_info
  running egg_info
  creating /tmp/pip-pip-egg-info-xd486t19/confluent_kafka.egg-info
  writing /tmp/pip-pip-egg-info-xd486t19/confluent_kafka.egg-info/PKG-INFO
  writing dependency_links to /tmp/pip-pip-egg-info-xd486t19/confluent_kafka.egg-info/dependency_links.txt
  writing requirements to /tmp/pip-pip-egg-info-xd486t19/confluent_kafka.egg-info/requires.txt
  writing top-level names to /tmp/pip-pip-egg-info-xd486t19/confluent_kafka.egg-info/top_level.txt
  writing manifest file '/tmp/pip-pip-egg-info-xd486t19/confluent_kafka.egg-info/SOURCES.txt'
  reading manifest file '/tmp/pip-pip-egg-info-xd486t19/confluent_kafka.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no files found matching 'test-requirements.txt'
  adding license file 'LICENSE.txt'
  writing manifest file '/tmp/pip-pip-egg-info-xd486t19/confluent_kafka.egg-info/SOURCES.txt'
  Preparing metadata (setup.py): finished with status 'done'
  Source in /tmp/pip-install-7g7fjlcr/confluent-kafka_9a3c7c6a7dbb4764b3a4d26658ee059b has version 2.5.0, which satisfies requirement confluent-kafka from https://files.pythonhosted.org/packages/94/51/2e8f4048edbcde0346047dcd5228985ddcd0ef4d518f878321c07cc1b39c/confluent-kafka-2.5.0.tar.gz
  Removed confluent-kafka from https://files.pythonhosted.org/packages/94/51/2e8f4048edbcde0346047dcd5228985ddcd0ef4d518f878321c07cc1b39c/confluent-kafka-2.5.0.tar.gz from build tracker '/tmp/pip-build-tracker-vwcy5rg0'
Created temporary directory: /tmp/pip-unpack-z2g2vjx1
Building wheels for collected packages: confluent-kafka
  Created temporary directory: /tmp/pip-wheel-rb_60q97
  Building wheel for confluent-kafka (setup.py): started
  Destination directory: /tmp/pip-wheel-rb_60q97
  Running command python setup.py bdist_wheel
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.9
  creating build/lib.linux-x86_64-3.9/confluent_kafka
  copying src/confluent_kafka/error.py -> build/lib.linux-x86_64-3.9/confluent_kafka
  copying src/confluent_kafka/deserializing_consumer.py -> build/lib.linux-x86_64-3.9/confluent_kafka
  ...............
  ...............
  ...............
  copying src/confluent_kafka/serialization/__init__.py -> build/lib.linux-x86_64-3.9/confluent_kafka/serialization
  creating build/lib.linux-x86_64-3.9/confluent_kafka/avro/serializer
  copying src/confluent_kafka/avro/serializer/message_serializer.py -> build/lib.linux-x86_64-3.9/confluent_kafka/avro/serializer
  copying src/confluent_kafka/avro/serializer/__init__.py -> build/lib.linux-x86_64-3.9/confluent_kafka/avro/serializer
  running build_ext
  building 'confluent_kafka.cimpl' extension
  creating build/temp.linux-x86_64-3.9
  creating build/temp.linux-x86_64-3.9/tmp
  creating build/temp.linux-x86_64-3.9/tmp/pip-install-7g7fjlcr
  creating build/temp.linux-x86_64-3.9/tmp/pip-install-7g7fjlcr/confluent-kafka_9a3c7c6a7dbb4764b3a4d26658ee059b
  creating build/temp.linux-x86_64-3.9/tmp/pip-install-7g7fjlcr/confluent-kafka_9a3c7c6a7dbb4764b3a4d26658ee059b/src
  creating build/temp.linux-x86_64-3.9/tmp/pip-install-7g7fjlcr/confluent-kafka_9a3c7c6a7dbb4764b3a4d26658ee059b/src/confluent_kafka
  creating build/temp.linux-x86_64-3.9/tmp/pip-install-7g7fjlcr/confluent-kafka_9a3c7c6a7dbb4764b3a4d26658ee059b/src/confluent_kafka/src
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -I/var/lang/include/python3.9 -c /tmp/pip-install-7g7fjlcr/confluent-kafka_9a3c7c6a7dbb4764b3a4d26658ee059b/src/confluent_kafka/src/Admin.c -o build/temp.linux-x86_64-3.9/tmp/pip-install-7g7fjlcr/confluent-kafka_9a3c7c6a7dbb4764b3a4d26658ee059b/src/confluent_kafka/src/Admin.o       
  In file included from /tmp/pip-install-7g7fjlcr/confluent-kafka_9a3c7c6a7dbb4764b3a4d26658ee059b/src/confluent_kafka/src/Admin.c:17:0:
  /tmp/pip-install-7g7fjlcr/confluent-kafka_9a3c7c6a7dbb4764b3a4d26658ee059b/src/confluent_kafka/src/confluent_kafka.h:23:10: fatal error: librdkafka/rdkafka.h: No such file or directory
   #include <librdkafka/rdkafka.h>
            ^~~~~~~~~~~~~~~~~~~~~~
  compilation terminated.
  error: command '/usr/bin/gcc' failed with exit code 1
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /var/lang/bin/python -u -c '
  exec(compile('"'"''"'"''"'"'
  # This is <pip-setuptools-caller> -- a caller that pip uses to run setup.py
  #
  # - It imports setuptools before invoking setup.py, to enable projects that directly
  #   import from `distutils.core` to work with newer packaging standards.
  # - It provides a clear error message when setuptools is not installed.
  # - It sets `sys.argv[0]` to the underlying `setup.py`, when invoking `setup.py` so
  #   setuptools doesn'"'"'t think the script is `-c`. This avoids the following warning:
  #     manifest_maker: standard file '"'"'-c'"'"' not found".
  # - It generates a shim setup.py, for handling setup.cfg-only projects.
  import os, sys, tokenize

  try:
      import setuptools
  except ImportError as error:
      print(
          "ERROR: Can not execute `setup.py` since setuptools is not available in "
          "the build environment.",
          file=sys.stderr,
      )
      sys.exit(1)

  __file__ = %r
  sys.argv[0] = __file__

  if os.path.exists(__file__):
      filename = __file__
      with tokenize.open(__file__) as f:
          setup_py_code = f.read()
  else:
      filename = "<auto-generated setuptools caller>"
      setup_py_code = "from setuptools import setup; setup()"

  exec(compile(setup_py_code, filename, "exec"))
  '"'"''"'"''"'"' % ('"'"'/tmp/pip-install-7g7fjlcr/confluent-kafka_9a3c7c6a7dbb4764b3a4d26658ee059b/setup.py'"'"',), "<pip-setuptools-caller>", "exec"))' bdist_wheel -d /tmp/pip-wheel-rb_60q97
  cwd: /tmp/pip-install-7g7fjlcr/confluent-kafka_9a3c7c6a7dbb4764b3a4d26658ee059b/
  Building wheel for confluent-kafka (setup.py): finished with status 'error'
  ERROR: Failed building wheel for confluent-kafka
  Running setup.py clean for confluent-kafka
  Running command python setup.py clean
  running clean
  removing 'build/temp.linux-x86_64-3.9' (and everything under it)
  removing 'build/lib.linux-x86_64-3.9' (and everything under it)
  'build/bdist.linux-x86_64' does not exist -- can't clean it
  'build/scripts-3.9' does not exist -- can't clean it
  removing 'build'
Failed to build confluent-kafka
Disabling truststore because Python version isn't 3.10+
Remote version of pip: 24.2
Local version of pip:  24.2
Was pip installed by pip? True
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (confluent-kafka)
Exception information:
Traceback (most recent call last):
  File "/var/lang/lib/python3.9/site-packages/pip/_internal/cli/base_command.py", line 105, in _run_wrapper
    status = _inner_run()
  File "/var/lang/lib/python3.9/site-packages/pip/_internal/cli/base_command.py", line 96, in _inner_run
    return self.run(options, args)
  File "/var/lang/lib/python3.9/site-packages/pip/_internal/cli/req_command.py", line 67, in wrapper
    return func(self, options, args)
  File "/var/lang/lib/python3.9/site-packages/pip/_internal/commands/install.py", line 434, in run
    raise InstallationError(
pip._internal.exceptions.InstallationError: ERROR: Failed to build installable wheels for some pyproject.toml based projects (confluent-kafka)
Removed build tracker: '/tmp/pip-build-tracker-vwcy5rg0'
`

**Comment by pranavrth** [comment]
I have understood the issue here. We recently changed the manylinux wheel generation to use `manylinux_2_28` distro and hence the tags that are present in the newer version is includes `manylinux_2_28`. Some of the older version of `python` and `pip` combination doesn't include this tag and hence the installation falls back to the installation from source distribution which is not advised during the normal installation as we provide the wheels. I will check on how to resolve this issue for the upcoming releases. Maybe using some older manylinux distribution should work. Will check this.

To **resolve** the above issues I think there are a few options here:

* Install latest version of Python and pip that you can use.
* Install pip using with the official way using [get-pip.py](https://pip.pypa.io/en/stable/installation/#get-pip-py)


If it is still using source distribution and failing then you can troubleshoot using the following guide:

**Issue 1** getting error on `#include <Python.h>` like the original logs of this post provided by @orlagetty.
``` /mnt/tmp/pip-install-blxssofn/confluent-kafka/src/confluent_kafka/src/confluent_kafka.h:18:10: fatal error: Python.h: No such file or directory
     #include <Python.h>
              ^~~~~~~~~~
    compilation terminated.
    error: command 'gcc' failed with exit status 1
```
**Cause** - This error is caused because you don't have devel package installed for python. 
**Solution** Install `python3-devel`/`python3-dev` package before trying to reinstall. [Refer: this answer for more help.](https://stackoverflow.com/a/21530768)


**Issue 2** Getting error on `#include <librdkafka/rdkafka.h>`
```/tmp/pip-install-7g7fjlcr/confluent-kafka_9a3c7c6a7dbb4764b3a4d26658ee059b/src/confluent_kafka/src/confluent_kafka.h:23:10: fatal error: librdkafka/rdkafka.h: No such file or directory
   #include <librdkafka/rdkafka.h>
            ^~~~~~~~~~~~~~~~~~~~~~
  compilation terminated.
  error: command '/usr/bin/gcc' failed with exit code 1
  error: subprocess-exited-with-error
```
**Cause** Installation from source distribution requires librdkafka to be installed before installing confluent_kafka.
**Solution** Installing librdkafka should fix this issue. [Please refer installation doc for librdkafka](https://github.com/confluentinc/librdkafka?tab=readme-ov-file#installation)

Please let me know if you have any concern.

**Comment by solucionesvaltech** [comment]
Hello!

I installed [librdkafka](https://github.com/confluentinc/librdkafka) in my machine and then restart the opened terminal and it works.

confluent-kafka==2.6.0

**Comment by AndreMCCarvalho** [comment]
To add to this thread, I'm facing installation errors on windows. I get the same error as above (librd) is missing.
I'm running python 3.13.0 and pip 24.3.1.
Wheels is installed. 
Anyone else facing this issue?

**Comment by MSeal** [comment]
That's very surprising that you see this on 3.13 with pip 24.3.1 -- I know this is an old issue but can you @AndreMCCarvalho give more details on the error and machine setup if you still are having problems? I'm going to close the issue as the original problem of old pip/python version is resolved by upgrading, but happy to engage if there's something further problematic with certain configurations.
